python/ray/train/v2/_internal/callbacks/state_manager.py

```diff
@@ -66,6 +66,8 @@ def after_controller_state_update(
         if previous_state._state_type == current_state._state_type:
             return
 
+        logger.info(f"Changed state from {previous_state} to {current_state}")
```

Comment:
nit: Use punctuation so it's easier to read.

```suggestion
        logger.info(f"Changed state from {previous_state} to {current_state}.")
```

Reply:
cc @justinvyu any thoughts on formatting? I had another idea that we could do something like:

`[State Transition] {previous} -> {current}`

but not sure if this would be better or worse 🤷

Reply:
Yeah I think the `->` is good

Should these be debug logs? so that users can eventually filter them out. These could possibly be spammy with the current `SCHEDULING -> RESCHEDLING -> SCHEDULING` loop when resources are not available.

Reply:
I think INFO should probably be fine?

If my understanding is correct, then the SCHEDULING -> RESCHEDULING -> SCHEDULING loop  will only happen when [this](https://github.com/ray-project/ray/blob/e2b19f7adb82db9c475ef7f7ed2aece2b91c8078/python/ray/train/v2/_internal/execution/controller/controller.py#L295-L299) is triggered, which will also emit an ERROR log anyways.

Reply:
Changed to the `->` style which I also prefer.
No strong opinions on info vs debug so I'll defer to you guys.

Reply:
This is how the log actually looks if you're curious

```
2025-05-27 18:24:59,116	INFO state_manager.py:69 -- [State Transition] RunningState(
    type=RUNNING
    is_terminal=False
    needs_new_run_attempt=False
) -> ErroredState(
    type=ERRORED
    is_terminal=True
    needs_new_run_attempt=False
    training_failed_error=Test error
).
```